lib/Hakyll/Core/Identifier.hs

```diff
@@ -32,6 +35,67 @@ import           GHC.Exts            (IsString, fromString)
 
 
 --------------------------------------------------------------------------------
+{- | A key data type to identify a compiled 'Hakyll.Core.Item.Item' in the 'Hakyll.Core.Store.Store'.
+Conceptually, it's a combination of a file path and a version name.
+The version is used only when a file is
+compiled within a rule using the 'version' wrapper function
+(the same source file
+can be compiled into several items in the store, so the version exists to distinguish
+them).
+Use functions like 'fromFilePath', 'setVersion', 'Hakyll.Core.Metadata.getMatches' to build an 'Identifier'.
+
+=== __Usage Examples__
+Normally, compiled items are saved to the store by 'Hakyll.Core.Rules.Rules' with an automatic, implicit identifier
+and loaded from the store by the user in another rule with a manual, explicit identifier.
+
+__Identifiers when using match__.
+Using 'Hakyll.Core.Rules.match' builds an implicit identifier that corresponds to the expanded, relative path
+of the source file on disk (relative to the project directory configured
+with 'Hakyll.Core.Configuration.providerDirectory'):
+
+@
+-- e.g. file on disk: 'posts\/hakyll.md'
+match "posts/*" $ do                                      -- saved with implicit identifier 'posts\/hakyll.md'
+    compile pandocCompiler
+
+match "about/*" $ do
+    compiledPost <- load (fromFilePath "posts/hakyll.md") -- load with explicit identifier
+    ...
+@
+Normally, the identifier is only explicitly created to pass to one the 'Hakyll.Core.Compiler.load' functions.
+
+__Identifiers when using create__.
+Using 'Hakyll.Core.Rules.create' (thereby inventing a file path with no underlying file on disk)
+builds an implicit identifier that corresponds to the invented file path:
+
+@
+create ["index.html"] $ do                            -- saved with implicit identifier 'index.html'
+    makeItem ("Hello world" :: String)
```

Comment:
Does not typecheck, missing `compile $`?

Reply:
Same for the examples with `load`, those are also actions in the Compiler monad.

Reply:
Thanks. I've correct all locations and hope i haven't missed one